dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,412 @@
+* Coinbase shall no longer carry the treasury subsidy transaction at TxOut[0].
+* Coinbase transaction version shall be 3.
+* The treasury subsidy value is subtracted from TxIn[0] value. An important consequence of this is that disapproved blocks no longer remove the per-block treasury payout.
+
+===Credit treasury===
+
+A user can send UTXOs to the treasury for a variety of reasons such as
+donations, overpaid contractor, moving old treasury balance to new treasury
+account etc.
+
+A treasury add transaction is identified by the following characteristics:
+* The wire message transaction version shall be 3.
+* There shall be at least one TxIn script.
+* There shall be one or two TxOut scripts.
+* All TxOut scripts shall be version 0.
+* TxOut[0] script must be <code>OP_TADD</code>.
+* TxOut[1] is optional if there is no change. If there is change it shall be a default <code>OP_SSTXCHANGE</code> script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0..N]||Default UTXO scripts||Normal UTXO scripts as you would find on normal transactions.
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be what is credited to the treasury.
+|-
+|TxOut[1]||[0]OP_SSTXCHANGE||Optional if there is no change.
+|}
+
+Additional consensus enforced characteristics:
+* A TAdd transaction shall not have a zero value for change.
+* There shall be no more than twenty TAdd transactions per block.
+
+===Treasury Expenditure Voting===
+
+Voting on valid TSpend transactions is done by piggy-backing on consensus
+voting transactions. The treasury votes are optionally appended to the
+consensus vote.
+
+A treasury vote is identified by the following characteristics:
+* The consensus vote format shall remain identical to the pre decentralized treasury agenda activation save for one optional output.
+* When treasury spend votes are appended the version of the transaction shall be 3.
+* The optional treasury spend vote output shall be the last output of the transaction.
+* The optional treasury spend votes output shall be a valid null script.
+* The null script shall be an <code>OP_RETURN</code> followed by a data push and the first two bytes of the data push shall be 'TV' (or in hex 0x54,0x56).
+* The remainder of the data length shall be 0 modulo of 32+1 (TSpend transaction hash + vote)
+* A TSpend transaction hash shall only occur once.
+* The vote shall be either <code>Yes = 0x01</code> or <code>No = 0x02</code> and all other values shall be rejected. This means the entire consensus vote shall fail.
+
+Since the format of the transaction remained the same only the optional data is
+described in the following tables. Note that there are two valid formats based
+on the number of treasury spend votes that are occurring simultaneously.
+
+Format 1
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[0]<code>OP_RETURN</code><br>[1]<code>OP_DATA_XX</code><br>[2..3]<code>'TV'</code><br>[4..]<code>treasury vote...</code>||Up to two (2) treasury votes can be encoded using this format.
```

Comment:
Same problems here regarding the script bits being hard to decipher with the offsets and stuff in there.

Same deal for the other cases below I won't comment on.